index.bs

```diff
@@ -1563,6 +1563,37 @@ Stats {#streaming-stats}
 TODO
 
 
+Protocol Extensions {#protocol-extensions}
+===================
+
+Open Screen agents may exchange extension messages that are not defined by this
+specification.  This could be used for experimentation, customization or other
+purposes.
+
+If an agent wishes to support extension messages, the author of the extension
+must register a capability ID and a range of message type keys in a public
+registry, whose location will be determined later.  Agents that accept the
+extension messages should include the corresponding capability ID in the
+`capabilities` field of its `agent-info` message.
+
+Capability IDs 1-999 are reserved for use by the Open Screen Protocol.
+Capability IDs 1000 and above are available for extensions.  See [[#appendix-b]]
+for legal ranges for extension message type keys.
+
+Note: This will prevent conflicts between multiple extension authors' capability
+IDs and message type keys.
+
+Agents must not send extension messages to another agent that has not advertised
```

Comment:
Should the spec call out what we do if an agent misbehaves?

Reply:
We already specify that if you receive a message that you don't know, close the connection (I think... if not, we should add it; but I think we already added it).

Reply:
Handling of unknown messages is discussed in the Messages section.  I added a cross reference in a Note here, just to help the reader determine know exactly what the agent needs to do.  But can remove it if folks think it's not necessary.